### PR TITLE
Only shuffle A and AAAA records on cache hits

### PR DESCRIPTION
**Pull request by cjellick**
We previously added logic to shuffle records when we got a cache hit. This
introduced a bug because we were shuffling CNAME records, whose order
should be maintained.

This change fixes that by only shuffling the sub section of records in the
response starting from the first A or AAAA record and going until the end.

### CODE REVIEW

**Comment by cjellick** [comment]
For https://github.com/rancher/rancher/issues/8669

File: answers.go [hunk 1]
```diff
@@ -281,10 +281,22 @@ func (answers *Answers) MatchingExact(qtype uint16, clientIp string, fqdn string
...
-		j := rand.Intn(i + 1)
+	min := 0
+	max := len(*items)
+	foundA := false
+	for i, record := range *items {
+		if !foundA && record.Header().Rrtype != dns.TypeA && record.Header().Rrtype != dns.TypeAAAA {
+			continue
+		}
+		foundA = true
+		if min == 0 {
```
Comment (vincent99) [inline answers.go]: I think this makes the shuffle unbalanced and you want just `j := i+rand.Intn(max-i)` every time so each entry is swapped exactly once.

**Comment by vincent99** [comment]
LGTM..

Nit: The test is going to break someday when `graph.facebook.com` changes...

**Comment by cjellick** [comment]
Yeah, I know. I guess its good enough for now and can be that way until it fails

**Pull request merged**